tests/Integration/ResourceStatusSystem/Test_EmailActionAgent.py

```diff
@@ -56,41 +45,37 @@ def setUp(self):
                                                            'description': 'A Policy that always returns Active'},
                             'Reason': 'AlwaysActive'}]
 
-    with patch.dict(os.environ):
-      os.environ.pop('DIRAC', None)
-      self.action = EmailAction(name, decisionParams, enforcementResult, singlePolicyResults)
+    self.action = EmailAction(name, decisionParams, enforcementResult, singlePolicyResults)
     self.action.log = gLogger
+    self.action.log.setLevel('DEBUG')
+
+    self.mockAM = MagicMock()
+    self.agent_m = importlib.import_module('DIRAC.ResourceStatusSystem.Agent.EmailAgent')
+    self.agent_m.AgentModule = self.mockAM
+    self.agent = self.agent_m.EmailAgent()
+    self.agent.am_getOption = self.mockAM
+    self.agent.log = gLogger
     self.agent.log.setLevel('DEBUG')
 
-    self.tc_mock = MagicMock()
-    self.tm_mock = MagicMock()
+  def tearDown(self):
+    pass
 
-  @classmethod
-  def tearDownClass(cls):
-    sys.modules.pop('DIRAC.ResourceStatusSystem.Agent.EmailAgent')
 
+class EmailActionAgent(TestEmailActionAgentTestCase):
 
-class EmailActionSuccess(TestCase):
+  def test__emailActionAgent(self):
 
-  def test__createTheDatabase(self):
-    res = self.action.run()
+    # clean up
+    res = self.rssClient.delete('ResourceStatusCache')
     self.assertTrue(res['OK'])
 
+    res = self.action.run()
+    self.assertTrue(res['OK'])
 
-class EmailAgentSuccess(TestCase):
-
-  def test__getData(self):
-    self.agent.diracAdmin = MagicMock()
     res = self.agent.execute()
     self.assertTrue(res['OK'])
 
-#############################################################################
-# Test Suite run
-#############################################################################
-
-
 if __name__ == '__main__':
-  suite = unittest.defaultTestLoader.loadTestsFromTestCase(TestCase)
-  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(EmailActionSuccess))
-  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(EmailAgentSuccess))
-  testResult = unittest.TextTestRunner(verbosity=2).run(suite)
+  suite = unittest.defaultTestLoader.loadTestsFromTestCase(TestEmailActionAgentTestCase)
+  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(EmailActionAgent))
+  unittest.TextTestRunner(verbosity=2).run(suite)
```

Comment:
```suggestion
  testResult = unittest.TextTestRunner(verbosity=2).run(suite)
  sys.exit(not testResult.wasSuccessful())
```

Reply:
Why is this change required?

Reply:
Done

Reply:
Running a python test file written, like this one, with `unittest` will produce exit code 0 even if the test fails. By adding
```
sys.exit(not testResult.wasSuccessful())
```
the test will exit with status code !=0 if the unit test is not successful.